File: 2025-05-MAY-GFG-POTD/May 15 - Substrings with same first and last characters/Solution.py
class Solution:
    def countSubstring(self, s: str) -> int:
        freq = [0] * 26
        for c in s:
            freq[ord(c) - ord('a')] += 1

        count = 0
        for f in freq:
            count += (f * (f + 1)) // 2

        return count

# Driver Code
if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        s = input()
        ob = Solution()
        result = ob.countSubstring(s)
        print(result)
        print("~")

File: 2025-05-MAY-GFG-POTD/May 15 - Substrings with same first and last characters/Solution.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int countSubstring(string s) {
        int freq[26] = {0};
        for (char c : s) {
            freq[c - 'a']++;
        }

        int count = 0;
        for (int f : freq) {
            count += (f * (f + 1)) / 2;
        }

        return count;
    }
};

int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;

        Solution ob;
        int result = ob.countSubstring(s);

        cout << result << endl;
        cout << "~" << endl;
    }
    return 0;
}
